RFC/src/RFC-0201_TariScript.md

```diff
@@ -495,20 +507,27 @@ $$
     \so = k_{Sa} - k_{Ob} \cdot \HU_b
 $$
 
-
 For the script hash, she provides the hash of a script that locks the output to Bob's public key, `PushPubkey(K_Sb)`.
 This script will only resolve successfully if the spender can provide a valid signature as input that demonstrates proof
 of knowledge of \\( k_{Sb} \\) which only Bob knows.
 
 Any base node can now verify that the transaction is complete, verify the signature on the script, and verify the script
 offset.
 
-For Bob to claim his commitment, \\( C_b \\) he requires the blinding factor, \\( k_b \\), and his private key for the script.
-
-In this case, the script hash is analogous to an address in Bitcoin or Monero. Bob's wallet can scan the blockchain
+For Bob to claim his commitment he will scan the blockchain for a known script hash because he knowns that the script will be `PushPubkey(K_Sb)` he can scan for that hash. In this case, the script hash is analogous to an address in Bitcoin or Monero. Bob's wallet can scan the blockchain
 looking for hashes that he would know how to resolve. For all outputs that he discovers this way, Bob would need to know
-who the sender is so that he can derive the shared secret.  This information can be obtained without needing to communicate
-with the sender, if she uses the offset public key, \\( K_{Ob} \\) as the key in deriving the Diffie-Hellman exchange.
+who the sender is so that he can derive the shared secret.
+
+When Bob's wallet spots a known hash he requires he requires the blinding factor, \\( k_b \\) and the value \\( v_b \\). First he uses Diffie-Hellman to calculate \\( k_b \\). 
+
+Bob calculates \\( k_b \\) as
+$$
+    k_b = K_{Ob} * {k_Sb}
+$$
+
+Next Bob's wallet calculates \\( k_{rewind} \\), using \\( k_{rewind} =  Hash(k_{b})\\) and uses that to rewind the Bulletproof to get the value \\( v_b \\). 
```

Comment:
We equate the private rewind key to the hash of the blinding factor. Bulletproof rewinding is done with the public rewind key, not the private rewind key.

Reply:
We actually need to convey two values, not only one. To be clear on the previous comment, Bulletproof rewinding is done with the two public rewind keys, not the private rewind keys.

From [Tari fork of Bulletproofs]( https://github.com/tari-project/bulletproofs/blob/a18a3d2c91cea4e5f999446a566a4b0af7c9e187/tests/range_proof.rs#L154)
``` rust
    // The two rewind keys can be shared with a trusted 3rd party, whom will be able to extract
    // the value of the commitment as well as the extra data
    let pub_rewind_key_1 =
        RistrettoPoint::from(&pvt_rewind_key * &RISTRETTO_BASEPOINT_TABLE).compress();
    let pub_rewind_key_2 =
        RistrettoPoint::from(&pvt_blinding_key * &RISTRETTO_BASEPOINT_TABLE).compress();
    // The rewind nonces are derived from the public rewind keys and the Pedersen commitment
    let rewind_nonce_1 = get_rewind_nonce_from_pub_key(&pub_rewind_key_1, &committed_value);
    let rewind_nonce_2 = get_rewind_nonce_from_pub_key(&pub_rewind_key_2, &committed_value);
    // Get secret nonces from the private keys - this emulates the owner of a wallet
    let blinding_nonce_1 = get_secret_nonce_from_pvt_key(&pvt_rewind_key, &committed_value);
    let blinding_nonce_2 = get_secret_nonce_from_pvt_key(&pvt_blinding_key, &committed_value);

    // Get Value Test 1 - provide correct rewind nonces
    let mut rewind_transcript = Transcript::new(b"Bulletproof-Rewind Test");
    assert_eq!(
        proof.rewind_single_get_value_only(
            &bp_gens,
            &mut rewind_transcript,
            &committed_value,
            64,
            &rewind_nonce_1,
            &rewind_nonce_2,
        ),
        Ok((confidential_value, proof_message))
    );
```